dask_jobqueue/sge.py

```diff
@@ -82,3 +84,42 @@ def __init__(self, queue=None, project=None, resource_spec=None, walltime=None,
         self.job_header = header_template % config
 
         logger.debug("Job script: \n %s" % self.job_script())
+
+    def start_workers(self, n=1, ta=True):
+        """
+        Start workers as a task array
+
+        Parameters
+        ----------
+        n : int
+            Total number of workers to start
+        ta : bool
+            If true, try to use task arrays
+        """
+        logger.debug('starting %s workers', n)
+
+        # Check if Task arrays can be used
+        ta = ta and bool(os.getenv('SGE_TASK_ID', None))
+
+        if ta:
+            logging.debug('Submission using task array')
+            _original_command_template = self._command_template
+            self._command_template = self._command_template.replace('${JOB_ID}', '${JOB_ID}.${SGE_TASK_ID}')
+
+            num_jobs = int(math.ceil(n / self.worker_processes))
+            header_template = self.job_header
+            self.job_header = header_template + '\n#$ -t 1-{}'.format(num_jobs)
```

Comment:
Yes I think that this too should go in `__init__()`

Reply:
Hi, I find this one a bit trickier, since `num_jobs` is defined when the `start_workers` function is called.
I could only think of passing `num_jobs` as
- a placeholder in the configuration (current solution):
    ```
    config = {'num_jobs_placeholder': '%(num_jobs)d'}`
    ...
    self.job_header = self.job_header % {'num_jobs' : num_jobs}
    ```
- an environmental variable to be picked up by `self.job_header`
- an attribute to substitute in `self.job_header`

Any other idea?

Reply:
You're right, no better idea. Just try to avoid as much as possible code duplication.